repose-aggregator/src/docs/asciidoc/recipes/docker.adoc

```diff
@@ -1,130 +1,147 @@
 = Docker
 
-With the advent of container technology, Repose can be fully encapsulated and run as a service.
-By bundling the environment with the software itself, deploying Repose becomes a much quicker, simpler process.
-Docker boasts security through isolation, and when run on platforms with native support, it requires very little overhead.
-To make everything as easy and versatile as possible, the Repose team will release Docker images on Docker Hub alongside every release of Repose.
+With the advent of container technology, *Repose* can be fully encapsulated and run as a service.
 
-== Docker
-To run Repose in Docker, there are just two basic steps:
+Even though the published *Repose Docker* image doesn't do a lot by default, this recipe will show you how to tap into some of that power hidden just below the surface.
 
-. Acquire a Repose Docker image.
-.. Download the Docker image. See: <<Docker Image>>
-.. Build the Docker image. See: <<Dockerfile>>
-. Run the Repose Docker image in a Docker container. See: <<Docker Container>>
```

Comment:
I have to say, I hate seeing some of this stuff go. I think what you have done here is write more of a recipe, which is absolutely appropriate, but I think there is still a place for more of this reference type material. I kind of want to stick it in an architecture page, but it's really more about Docker than it is Repose.

I wonder if the old page wouldn't work well as a README for the Docker project. I think it does a fair job detailing how to use Docker and build Repose with it in a structured way. Repose users probably don't want or need such information, but I think it does have value for developers.

Perhaps the team needs to have a discussion about user vs. developer documentation. The idea of putting developer documentation in the project as READMEs and user documentation on the site is starting to grow on me.

Thoughts?

Reply:
I do like the idea of putting `README`'s wherever they are needed.
It keeps the tribal knowledge from being lost and reinforces that others should be doing the same.

What all would you want to be in that document?

Reply:
Also, this block was just moved into another section.

Reply:
Yeah, I went back and wrote this comment at the end. I know some of the stuff was migrated, but other stuff was removed (that is, a search for it only yielded one result, which was highlighted with red). Of course, some of the removed stuff may have just been changing to be worded in a different way, but some of it is actually gone.

To me, developer documentation would basically provide a quick reference to the concepts and tools used to do what we do with Docker.

So one of the major changes is the organization. This recipe groups information by the goal is helps to achieve. The old page grouped information by what Docker concept it fell under. The recipe approach is great for a user who just wants to achieve something, while the old page is great for a developer to quickly find a way to get something working. So if I, as a developer, wanted to know how to inspect the state of my container, having a section named `Inspection` makes it easy to find that information.

Which brings me to the `Inspecting` section. It is just plain gone. That has certainly been useful to me while working with Docker, so I believe it is probably useful to others, as well. 

Or the bit about log files. Yes, that section needs to be updated because mounting the log directory is not the recommended way of reading logs, but it's nice to have a quick reference on how to read the logs. I know that's on the quick start page, so that may just be my organizational complaint again... although, developer documentation could provide multiple options for reading logs (e.g., mounting the log directory as a volume vs using Docker log management) even though we would not want to tell users to mount a log directory. In this specific example, it may be easier just to mount the log directory if, say, we are debugging with a set of configs handed to us by some other team.

I also liked the `Docker` section that told you, at a high level, what is necessary to run Repose in Docker.

So yeah, different ways of presenting (sometimes different) information to different audiences. That's my argument for keeping most of the old page around (although perhaps with some needed updates) as developer documentation.

Reply:
This was added to the new developer docs in almost the same format.